src/launchpad/analyzers/apple.py

```diff
@@ -120,12 +120,26 @@ def analyze(self, artifact: AppleArtifact) -> AppleAnalysisResults:
             )
             treemap = treemap_builder.build_file_treemap(file_analysis)
 
+        insights: AppleInsightResults | None = None
+        if not self.skip_insights:
+            logger.info("Generating insights from analysis results")
+            insights_input = InsightsInput(
+                app_info=app_info,
+                file_analysis=file_analysis,
+                binary_analysis=binary_analysis,
+                treemap=treemap,
+            )
+            insights = AppleInsightResults(
+                duplicate_files=DuplicateFilesInsight().__call__(insights_input),
```

Comment:
Got rid of the prior "fanciness" that was hoping to execute the insights in a loop. Now simply sets a value.